nemo_aligner/data/nlp/datasets.py

```diff
@@ -306,9 +308,11 @@ def __getitem__(self, idx):
         assert reject[0:prompt_len] == prompt, "the tokenizer for DPO has merged tokens between prompt and response"
 
         max_curr_seq_len = max(chosen_len, reject_len)
-        assert (
-            max_curr_seq_len <= self.seq_length
-        ), "tokenized text exceeds max seq len! truncate your data in preprocessing prior to DPO training"
+        if max_curr_seq_len > self.seq_length:
+            logging.warning(
+                f"WARNING: Tokenized text exceeds max seq length ({max_curr_seq_len} vs {self.seq_length})."
+                + f"The example will be ignored."
+            )
```

Comment:
im ok with this change, but if we're going to do this how about we just do filtering in the `__init__` of this dataset?
this way we don't have to hack the `__getitem__`. Our datasets are usually not big enough for this to matter in the `__init__`

Reply:
Actually, it's more efficient to do the hacking in `__getitem__` because that's the part which gets multiprocessed, but `__init__` does not. However, the current approach is rather inefficient because it takes an entire seq_length of tokens even though their labels are all negative.

Reply:
I haven't made any updates in this thread for the latest commit. In principle, it is more efficient if we do it in __init__ at the cost of a less prettier code.

Reply:
agree that `__getitem__` is better since we can run it on many workers but code wise I think we should do this filtering in init and error if there's no valid samples. Then this prevents us from having to mask things, and random samples that raise this warning that might slip through

Reply:
I suggest that we leave the init filtering to a separate PR since it might need more careful discussions.

Reply:
yeah that's fine cc @terrykong